userspace/libscap/scap.h

```diff
@@ -144,6 +145,20 @@ typedef struct scap_stats
 	uint64_t n_tids_suppressed; ///< Number of threads currently being suppressed.
 }scap_stats;
 
+/*!
+  \brief Statistics about an in progress capture (including counters and libbpf stats, compare to `bpftool prog show` CLI).
+*/
+typedef struct scap_stats_v2
+{
+	/* Metadata */
+	char name[STATS_NAME_MAX];
+	bool valid;
+	uint32_t flags;
+	/* Stats values */
+	uint64_t u64value;
```

Comment:
Values can be represented as a union so that memory usage can be optimized. I wouldn't expect a stats value to change type at runtime. This would need to be paired with an enum describing the supported types and use that in this struct to define the type of the value.